### PR TITLE
perf: optimize halo2 prover (2)

### PR DESCRIPTION
**Pull request by chokobole**
# Description

1. Halo2 holds the polynomials in both coefficient and evaluation form. Considering that the polynomials in coefficient form are needed after y, we can delay IFFT until phase y. This can help reduce memory pressure.
2. The compressed pairs are needed to construct grand product polys. The h poly and evals are needed to construct combined h poly. After they are used, this commit immediately releases their memory.
3. Since the conversion from bucket to affine point requires an inverse operation, `BatchNormalize()` is used to reduce the number of required inverse operations. This commit tries to use `BatchNormalize()` on every commitment after phase theta.

### CODE REVIEW

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by ashjeong — COMMENTED** [review]
https://github.com/kroma-network/tachyon/pull/294/commits/0cdb82c56a818617d4b3d77a5d2174465083f69a
change message to match edited pr description (changed "y phase," "reducing," "releases memory," "operation," & last sentence)

File: tachyon/zk/lookup/halo2/compress_expression.h [hunk 1]
```diff
@@ -4,8 +4,8 @@
 // can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
```
[the remark this answers is not in the document]
Comment (chokobole) [reply tachyon/zk/lookup/halo2/compress_expression.h:4]: I'll try!

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by Iamkeshy — COMMENTED** [review]
All

**Pull request merged**